pyccel/ast/numpyext.py

```diff
@@ -652,8 +656,9 @@ def __init__(self, shape, fill_value, dtype=None, order='C'):
 
         # Cast fill_value to correct type
         if fill_value and not isinstance(fill_value, Nil):
```

Comment:
Could we provide `Nil` with a `__bool__` magic method? If so we would just write
```python
if fill_value:
```

Reply:
It already has one. This ought to have been changed when it was added I think. Fixed in 47a2297